<launch>

 <include file="$(find xycar_motor)/launch/xycar_motor.launch" />
 <include file="$(find xycar_ultrasonic)/launch/xycar_ultrasonic_noviewer.launch" />
 <include file="$(find obstacle_detector)/launch/final_pid.launch" />

<!-- image_raw -->
  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
    <param name="video_device" value="/dev/videoCAM" />
    <param name="image_width" value="640" />
    <param name="image_height" value="480" />
    <param name="exposure" value="50" />
    <param name="autoexposure" value="false" />
    <param name="pixel_format" value="yuyv" />
    <param name="camera_frame_id" value="usb_cam" />
    <param name="io_method" value="mmap"/>
    <param name="camera_info_url" value="file://${ROS_HOME}/camera_info/head_camera.yaml"/>
  </node>
  
<!-- image_rect -->
  <node name="image_proc" pkg="image_proc" type="image_proc" ns="usb_cam" /> 
  <node name="image_rect_color" pkg="image_view" type="image_view">

    <arg name="marker_size" default="5.2" />
    <arg name="max_new_marker_error" default="0.08" />
    <arg name="max_track_error" default="0.2" />
    <arg name="cam_image_topic" default="/usb_cam/image_rect" />
    <arg name="cam_info_topic" default="/usb_cam/camera_info" />
    <arg name="output_frame" default="usb_cam" />
   
    <node name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen">
        <param name="marker_size"           type="double" value="$(arg marker_size)" />
        <param name="max_new_marker_error"  type="double" value="$(arg max_new_marker_error)" />
        <param name="max_track_error"       type="double" value="$(arg max_track_error)" />
        <param name="output_frame"          type="string" value="$(arg output_frame)" />

        <remap from="camera_image"  to="$(arg cam_image_topic)" />
        <remap from="camera_info"   to="$(arg cam_info_topic)" />
    </node>
    
    <node name="auto_drive" pkg="final_drive" type="main.py" output="screen" />
</launch>
